stamina-core/src/main/scala/stamina/migrations.scala

```diff
@@ -36,40 +36,71 @@ package migrations {
+   * app that actually saves data in new format.
+   * Note that `backFrom[V]` is effective only when
+   * it is called as the last one in call chain when defining
+   * `Migrator[R, V]`
    *
    * @example Using the json implementation:
    * {{{
    * val p = persister[CartCreated, V3]("cart-created",
    *   from[JsValue, V1]
    *     .to[V2](_.update('cart / 'items / * / 'price ! set[Int](1000)))
    *     .to[V3](_.update('timestamp ! set[Long](System.currentTimeMillis - 3600000L)))
+   *     .backFrom[V4](_.update('cart / 'items / * / 'name ! set[String]("unknown")))
    * )
    * }}}
    *
    *  @tparam R The type of raw data being migrated. In the JSON implementation this would be `JsValue`.
-   *  @tparam V The "current" version of this Migrator, i.e. it can migrate values from V1 to this version or any version in between.
+   *  @tparam V The "current" version of this Migrator, i.e. it can migrate values from V1 to this version or any version in between
+    *            and optionally from next version back to this one.
    */
-  class Migrator[R, V <: Version: VersionInfo] private[stamina] (migrations: Map[Int, Migration[R]] = Map.empty) {
-    def canMigrate(fromVersion: Int): Boolean = migrations.contains(fromVersion)
+  class Migrator[R, V <: Version : VersionInfo] private[stamina](migrations: Map[Int, Migration[R]] = Map.empty, backwardMigration: Option[Migration[R]] = None) {
+    def canMigrate(fromVersion: Int): Boolean = migrations.contains(fromVersion) || (backwardMigration.isDefined && fromVersion == Version.numberFor[V] + 1)
 
     def migrate(value: R, fromVersion: Int): R = {
-      migrations.get(fromVersion).map(_.apply(value)).getOrElse(
-        throw UndefinedMigrationException(fromVersion, Version.numberFor[V])
-      )
+      val thisVersion = Version.numberFor[V]
+      if (fromVersion <= thisVersion) {
+        migrations.get(fromVersion).map(_.apply(value)).getOrElse(
+          throw UndefinedMigrationException(fromVersion, thisVersion)
+        )
+      } else if (fromVersion == thisVersion + 1) {
+        backwardMigration.map(_.apply(value)).getOrElse(
+          throw UndefinedMigrationException(fromVersion, thisVersion)
+        )
+      } else {
+        throw UndefinedMigrationException(fromVersion, thisVersion)
+      }
     }
 
-    def to[NextV <: Version: VersionInfo](migration: Migration[R])(implicit isNextAfter: IsNextVersionAfter[NextV, V]) = {
+    def to[NextV <: Version : VersionInfo](migration: Migration[R])(implicit isNextAfter: IsNextVersionAfter[NextV, V]): Migrator[R, NextV] = {
       new Migrator[R, NextV](
-        migrations.mapValues(_ && migration) + (Version.numberFor[NextV] → identityMigration[R])
+        migrations.mapValues(_ && migration) + (Version.numberFor[NextV] → identityMigration[R]),
+        None
+      )
+    }
+
+    def backFrom[NextV <: Version : VersionInfo](migration: Migration[R])(implicit isNextAfter: IsNextVersionAfter[NextV, V]): Migrator[R, V] = {
```

Comment:
Hoiw about renaming this method to `andBackFrom`? This way it will gramaticall point out that this method should be called only once and at the end of the `to` method calls: 
```from[V1](...).to[V2](...).to[V3].andBackFrom[V4](...)```
Or even:
```from[V1](...).andBackFrom[V4](...)```

While this:
```from[V1](...).to[V2](...).andBackFrom[V3](...).to[V3](...)```
and this:
```from[V1](...).to[V2](...).andBackFrom[V3](...).andBackFrom[V4](...)```
would automatically look suspicious in code and would be easier to catch in code reviews.